hash_practice/exercises.py

```diff
@@ -1,19 +1,50 @@
 
-def grouped_anagrams(strings):
+def grouped_anagrams(words):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n^2)
+        Space Complexity: O(n)
     """
-    pass
+
+    array = []
+    dic = {}
+    for word in words:
+        char_in_words_sum = 0
+        for char in word:
+            char_in_words_sum += ord(char)
+        if char_in_words_sum in dic:
+            dic[char_in_words_sum].append(word)
+        else:
+            dic[char_in_words_sum] = [word]
+            # print(dic)
+    for key in dic:
+        array.append(dic[key])
+
+    return array
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
-    pass
+
+    dic = {}
+    results_list = []
+    for element in nums:
+        if element in dic:
+            dic[element] += 1
+        else:
+            dic[element] = 1
+    dic_values_list = (list(dic.values()))

```

Comment:
Looks like some extra parentheses here around `list(dic.values())`. They don't really break anything, but usually it's worth dropping them just to maintain a clean style.